src/video/ogc/SDL_ogcgxcommon.c

```diff
@@ -40,12 +41,32 @@ static const f32 tex_pos[] __attribute__((aligned(32))) = {
     1.0,
 };
 
+bool OGC_get_aspect_ratio_dimensions(float *w, float *h)
```

Comment:
It's not really needed to use two floats here. Either do

```suggestion
bool OGC_get_aspect_ratio(float *ratio)
```

or

```suggestion
bool OGC_get_aspect_ratio_dimensions(int *w, int *h)
```

Reply:
Also, it's better to cache the result of this function, since we don't expect the variable to change during run-time. If you decide to go with a single float, caching could be achieved like this:

```c
static bool OGC_get_aspect_ratio_real(float *ratio) {
    ...implementation...
}

bool OGC_get_aspect_ratio(float *ratio) {
    static float cached_ratio = -1.0f;
    if (cached_ratio < 0) {
        bool has_ratio = OGC_get_aspect_ratio_real(&cached_ratio);
        if (!has_ratio) cached_ratio = 0.0;
    }
    *ratio = cached_ratio;
    return ratio != 0.0;
}
```
You could even remove the boolean value, and make the function return the float directly; then, if it's 0.0, it means that no ratio has been set.

Reply:
The reason I did it this way was just for accuracy, I feared rounding errors if I didn't use 2 floats
If I just return the ratio it'll need to be a float tough

Reply:
made it a float, will cache it when I get the chance